src/Workspaces/CSharp/Portable/Classification/ClassificationHelpers.cs

```diff
@@ -118,6 +118,46 @@ private static string GetClassificationForIdentifier(SyntaxToken token)
             {
                 return ClassificationTypeNames.TypeParameterName;
             }
+            else if (token.Parent is MethodDeclarationSyntax methodDeclaration && methodDeclaration.Identifier == token)
+            {
+                // TODO: Add proper extension method check - I'm sure there is some syntax based check somewhere burried in roslyn but I couldn't find it.
+                bool isExtensionMethod =
+                    (methodDeclaration.Parent as TypeDeclarationSyntax)?.Modifiers.Any(SyntaxKind.StaticKeyword) == true &&
+                    methodDeclaration.Modifiers.Any(SyntaxKind.StaticKeyword) &&
+                    methodDeclaration.ParameterList.Parameters.FirstOrDefault()?.Modifiers.Any(SyntaxKind.ThisKeyword) == true;
```

Comment:
basically, my point is this: just check the parameter.  you don't need the other checks as they're superfluous.

Reply:
Is there an extension method check somewhere ?

Reply:
not that i know of.